atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/TracingQueryRunner.java

```diff
@@ -75,7 +78,7 @@ public TracingQueryRunner(Logger log, TracingPrefsConfig tracingPrefs) {
 
     private boolean shouldTraceQuery(Set<TableReference> tableRefs) {
         for (TableReference tableRef : tableRefs) {
-            if (tracingPrefs.shouldTraceQuery(tableRef.getQualifiedName())) {
+            if (tracingPrefs.get().shouldTraceQuery(tableRef.getQualifiedName(), random)) {
```

Comment:
Can we just use `ThreadLocalRandom`?
```suggestion
            if (tracingPrefs.get().shouldTraceQuery(tableRef.getQualifiedName(), ThreadLocalRandom.current())) {
```

Reply:
From my end, no strong preferences, although as noted I'd much rather we do the random code in here (and thus don't need to pass a `Random` around).

Reply:
I'm currently just preserving existing behavior where a single instance of Random was used for all threads. Can we make behavior changes separately?